submissions/SamVal007/Friends_APP/script.js

```diff
@@ -0,0 +1,127 @@
+const MENU_FORMS = document.getElementById('menu_forms');
+const SEARCH_FIELD = document.getElementById('searchField');
+const URL = 'https://randomuser.me/api/?results=40';
+const btn = document.querySelector('.resetButton');
+console.log(btn);
+
+let persons = [];
+let displayList = [];
+const putPersonsInDisplayList = () => {displayList = persons;};
+
+function makeFriendList() {
+  let response = fetch(URL)
+
+    .then(
+      successResponse => {
+        if (successResponse.status != 200) {
+          return null;
+        } else {
+          return successResponse;
+        }
+      })
+
+    .then(data => data.json())
+
+    .then(function (data) {
+      persons = data.results;
+      putPersonsInDisplayList();
+    })
+
+    .then(function () { // without this .then i dodnt how to load displayCards() on DOMContentLoaded
+      displayCards();
+    })
+
+    .catch(function (error) {
+      console.log('We have an error: ', error);
+      alert('Please, check your internet connection and try again');
+    })
+}
+
+function getTemplate(person) {
+  return `<div class="card">
+            <div class="picture">
+              <img src="${person.picture.large}">
+            </div>
+            <h1>${person.name.last} ${person.name.first}</h1>
+            <div class="cardInfo">
+              <p><span class="text_title">Age:</span> ${person.dob.age}</p>
+              <p><span class="text_title">Gender:</span> ${person.gender}</p>
+              <p><span class="text_title">Phone:</span> ${person.phone}</p>
+              <p><span class="text_title">Country:</span> ${person.location.country}</p>
+              <p><span class="text_title">City:</span> ${person.location.city}</p>
+            </div>
+          </div>`
+}
+
+function displayCards() {
+  CONTAINER.innerHTML = '';
+  let cardMarkup = '';
+  cardMarkup += displayList.reduce((accumulator, currentValue) => accumulator.concat(getTemplate(currentValue)), '');
+  CONTAINER.insertAdjacentHTML('afterbegin', cardMarkup);
+
+}
+////// Sorting Logic ////
+const compareAge = (firstFriend, secondFriend) => {
+  return firstFriend.dob.age - secondFriend.dob.age;
+};
+
+const compareName = (firstFriend, secondFriend) => {
+  return firstFriend.name.last < secondFriend.name.last ? -1 : 1;
+};
+
+const ageSorters = {
+  ageLow: () => persons.sort((a, b) => compareAge(b, a)),
+  ageHigh: () => persons.sort(compareAge)
+};
+
+const nameSorters = {
+  nameDesc: () => persons.sort((a, b) => compareName(b, a)),
+  nameAsc: () => persons.sort(compareName)
+};
+const filtrByMale = () => {
+  displayList = persons.filter(el => el.gender === 'male');
+};
+const filtrByFemale = () => {
+  displayList = persons.filter(el => el.gender === 'female');
+};
+//////////////////////////////////////////////////////////////
+
+
+document.addEventListener('DOMContentLoaded', function () {
+  makeFriendList();
+  //displayCards(); //I dodnt know why, but if put displayCards() here -> it doesnt work
```

Comment:
So you need to found out :D Or you need my help?

Reply:
It seems, now it works as we want with single responsibility!
But I watched one video about **pure functions** and understood, that this my code is smelling a little)) Right?:) 
I should make it without using GLOBAL arrays ` let persons = []; let displayList = []; `. Right?